src/OrchardCore.Modules/OrchardCore.Media.AmazonS3/Startup.cs

```diff
@@ -0,0 +1,170 @@
+﻿using System;
+using System.IO;
+using System.Linq;
+using System.Text;
+
+using Amazon;
+using Amazon.Runtime;
+using Amazon.S3;
+
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.DependencyInjection.Extensions;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+using OrchardCore.Environment.Shell;
+using OrchardCore.Environment.Shell.Configuration;
+using OrchardCore.FileStorage;
+using OrchardCore.Media.Core;
+using OrchardCore.Media.Events;
+using OrchardCore.Modules;
+using OrchardCore.FileStorage.AmazonS3;
+using OrchardCore.Media.Core.Events;
+
+namespace OrchardCore.Media.AmazonS3;
+
+public class Startup : Modules.StartupBase
+{
+    private readonly ILogger _logger;
+    private readonly IShellConfiguration _configuration;
+    private readonly IWebHostEnvironment _webHostEnvironment;
+
+    public Startup(IShellConfiguration configuration,
+        IWebHostEnvironment webHostEnvironment,
+        ILogger<Startup> logger)
+        => (_configuration, _webHostEnvironment, _logger)
+            = (configuration, webHostEnvironment, logger);
+
+    public override void ConfigureServices(IServiceCollection services)
+    {
+        var storeOptions = new AwsStorageOptions().BindConfiguration(_configuration);
+
+        var validationErrors = storeOptions.Validate().ToList();
+        var stringBuilder = new StringBuilder();
+
+        if (validationErrors.Any())
+        {
+            foreach (var error in validationErrors)
+            {
+                stringBuilder.Append(error.ErrorMessage);
+            }
+
+            if (_webHostEnvironment.IsDevelopment())
+            {
+                _logger.LogInformation(
+                    $"S3 Media configuration validation failed: {validationErrors} fallback to File storage", stringBuilder);
+            }
+            else
+            {
+                _logger.LogError(
+                    $"S3 Media configuration validation failed with errors: {validationErrors} fallback to File storage", stringBuilder);
+            }
+        }
+        else
+        {
+            _logger.LogInformation(
+                $"Starting with S3 Media Configuration. {System.Environment.NewLine} BucketName: {storeOptions.BucketName}, {System.Environment.NewLine} BasePath: {storeOptions.BasePath}");
```

Comment:
Parameterize the logging template.

Reply:
This is still not addressed :). What I'm telling is, don't add variables in the string, but supply them as parameters, like you do on line 51.